arch/arc/core/thread.c

```diff
@@ -218,12 +220,14 @@ FUNC_NORETURN void _arch_user_mode_enter(k_thread_entry_t user_entry,
 	_current->arch.k_stack_top = _current->arch.priv_stack_start;
 	_current->arch.k_stack_base = _current->arch.priv_stack_start +
 				_current->arch.priv_stack_size;
-	_current->arch.u_stack_top = _current->stack_info.start;
+	/* +4 bytes to reserve place for errno */
+	_current->errno_location = (int *)_current->stack_info.start;
```

Comment:
Required? Isn't errno_location already initialized in the generic kernel/thread.c ?